From: tcamuso@redhat.com <tcamuso@redhat.com>
Date: Thu, 28 Aug 2008 17:43:42 -0400
Subject: [misc] hpilo: cleanup device_create for RHEL-5.3
Message-id: 20080828214415.7055.52604.sendpatchset@dhcp-100-2-186.bos.redhat.com
O-Subject: [PATCH 3/3]hpilo driver - backport changes from upstream 2.6.27
Bugzilla: 437212

Patch 3 of 3
============
A patch set to update hpilo driver from version 0.2 to version 0.5, which
is the version in upstream 2.6.27.

Patch Description
=================
This patch removes the new upstream "void *drvdata" argument from the call to
device_create().

Bugzilla
========
https://bugzilla.redhat.com/show_bug.cgi?id=437212

Upstream Status
===============
commit f38954c93c4a548f55d73ac5c1cf5e7f4023bb6c

Test Status
===========
David Altobelli built and tested this patch set against 2.6.18-105
for functionality and regression, and the driver performed as
expected with no regressions.

Summary
=======
     drivers/misc/hpilo.c |    2 +-
     1 files changed, 1 insertions(+), 1 deletions(-)

    Signed-off-by: Tony Camuso
    Signed-off-by: David Altobelli

diff --git a/drivers/misc/hpilo.c b/drivers/misc/hpilo.c
index 05e2982..cc72902 100644
--- a/drivers/misc/hpilo.c
+++ b/drivers/misc/hpilo.c
@@ -688,7 +688,7 @@ static int __devinit ilo_probe(struct pci_dev *pdev,
 	for (minor = 0 ; minor < MAX_CCB; minor++) {
 		struct device *dev;
 		dev = device_create(ilo_class, &pdev->dev,
-				    MKDEV(ilo_major, minor), NULL,
+				    MKDEV(ilo_major, minor),
 				    "hpilo!d%dccb%d", devnum, minor);
 		if (IS_ERR(dev))
 			dev_err(&pdev->dev, "Could not create files\n");
